chapters/functions.tex

```diff
@@ -2834,6 +2834,7 @@ \subsection{External Objects}\label{external-objects}
   The {\lstinline!constructor!} shall have exactly one output argument in which the constructed instance derived from {\lstinline!ExternalObject!} is returned.
   The arguments to the constructor must not -- directly nor indirectly -- depend on the external object being constructed.
   The constructor shall initialize the object, and must not require any other calls to be made for the initialization to be complete (e.g., from an initial algorithm or initial equation).
+  The constructor shall return a pointer to the object, returning a null pointer signifies an error.
```

Comment:
If we want to stress that the return value is a pointer, I think that would belong in the sentence about _exactly one output argument_.  Here, the significant part is the interpretation of a null pointer result.

We need to be careful about the distinction between being an error (that is, illegal according to the specification) and behaving as if `ModelicaError` was used.  I think we can take the [position suggested by @HansOlsson](https://github.com/modelica/ModelicaSpecification/issues/3500#issuecomment-2057189696) as a starting point for the discussion:
```suggestion
  To indicate failure, the \lstinline!constructor! may return a null pointer as an alternative to invoking \lstinline{ModelicaError}.
```

Reply:
An alternative which I find quite appealing myself would be to instead require the use of {ModelicaError}:
```suggestion
    The \lstinline!constructor! must not return a null pointer; to indicate failure it needs to invoke \lstinline[language=C]!ModelicaError! or one of the other non-returning functions in \label{error-reporting-utility-functions}.
```

Reply:
One upside with the null pointer alternative is that it's actually theoretically possible to integrate an existing C library without writing a single line of C code (since returning a null pointer is a quite common way to indicate failure). Although I don't know how useful that is in practice.

Reply:
We can discuss it, but I agree that just treating it as error makes the most sense.